x-pack/plugins/threat_intelligence/public/app.tsx

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CoreStart } from '@kbn/core/public';
+import React, { VFC } from 'react';
+import { IndicatorsPage } from './modules/indicators/indicators_page';
+import { KibanaContextProvider } from './hooks/use_kibana';
+
+export interface AppProps {
+  services: CoreStart;
+}
+
+export const App: VFC<AppProps> = ({ services }) => {
+  return (
+    <KibanaContextProvider services={services}>
```

Comment:
That's great, I like that you have your own `KibanaContextProvider`, I would only add that instead of passing the `services` from the component consumer (Security), it would probably be better to use the core `services` from your own (threat_intelicence) plugin `start` function. This way you will manage your own services and dependencies instead of relying on the Security dependencies.

Reply:
that makes thanks, I fixed (hopefully properly), thank you!

Reply:
Not exactly, the proper way would be to do something like:
```
export const createAppComponent = (services: CoreStart) => {
  const App = () => (
    <KibanaContextProvider services={services}>
      <IndicatorsPage />
    </KibanaContextProvider>
  );
  return App;
};

```
and then in your _x-pack/plugins/threat_intelligence/public/plugin.tsx_, the `start` function would have to pass the services, which are composed of the core services + the plugins that you required in your own kibana.json. It would look like this:
```
  public start(core: CoreStart, plugins: PluginsStart): ThreatIntelligencePluginStart {
    const App = createAppComponent({ ...core, ...plugins });
    return { getComponent: () => App };
  }
```
The `PluginsStart` will have to be defined, I see currently you already have some _requiredPlugins_ in your kibana.json.

It is important to note that the `KibanaContextProvider` merges the parent context services (Security) inside the children's context services (Theat Intel), so, even doing these changes, you'll still have all the Security services available when you do `useKibana().services` inside your context.
As you might guess, this change has no real impact right now. But doing so we decouple the Threat intel dependencies from Security, and you won't have problems if any other solution wants to import your components. Also, if you need to add a plugin dependency that Security does not have (e.g. `charts`), you will be able to add it on your own _Threat Intel_ kibana.json. With the current approach, you would have to add it in the _Security_ kibana.json.

Hope it helps

Reply:
hmm I think I see, let me give it a shot. Thanks for the detailed answer! 😄